### ISSUE TITLE
"flake8-boolean-trap.extend-allowed-calls" option does not work in my case

### ISSUE THREAD

**Issue by clara-bordes** [astral-sh/ruff#11264]
I am actually working on fixing the rule **boolean-positional-value-in-call (FBT003)** on a code repository.

I need to allow this rule for some functions. For that, at the end of my config file "pyproject.toml", I have added:

```
[lint.flake8-boolean-trap]
extend-allowed-calls = ["LibName.className.functionName"]
```

My issue here is that in my code repository, we call the function "LibName.className.functionName" through a variable, as following:

```
from LibName import ClassName

with ClassName("[Description]") as variable:
       variable.functionName(...)
```

On such code, the rule FBT003 is **still applied** in spite of the "extend-allowed-calls" option.

Is the "extend-allowed-calls" option only works for direct function calls?

**Comment by charliermarsh** [comment]
Yes, that setting doesn't support resolution to the base type. It won't trace `variable` back to `LibName.ClassName`.

**Comment by charliermarsh** [comment]
I'd suggest just disabling the rule, but we'll probably support this at some point in the future.

**Comment by flying-sheep** [comment]
would be nice to support wildcards in the meantime. E.g. I think all methods called like `.fill(True)` are probably fine.

**Comment by andrei-korshikov** [comment]
> would be nice to support wildcards in the meantime

+1 on wildcards (regexes?). My use case:

```python
import pycurl

curl = pycurl.Curl()
curl.setopt(pycurl.VERBOSE, True)
```

[pycurl.Curl.setopt()](http://pycurl.io/docs/latest/curlobject.html#pycurl.Curl.setopt) is positional-only (takes no keyword arguments). It should be part of #3247 when type resolution will be available.

**Comment by danra** [comment]
Ran into this as well trying to whitelist the canonical `torch.Tensor.requires_grad_`
